hazelcast-client/src/test/java/com/hazelcast/client/ClientServiceTest.java

```diff
@@ -436,4 +436,102 @@ public void stateChanged(LifecycleEvent event) {
         hazelcastInstance.shutdown();
         assertOpenEventually(countDownLatch, ClusterListenerSupport.TERMINATE_TIMEOUT_SECONDS);
     }
+
+    @Test
+    public void testClientListener_fromConfig() {
+        Config config = new Config();
+        final CountDownLatch latch = new CountDownLatch(2);
+        ListenerConfig listenerConfig = new ListenerConfig(new ClientListener() {
+            @Override
+            public void clientConnected(Client client) {
+                latch.countDown();
+            }
+
+            @Override
+            public void clientDisconnected(Client client) {
+                latch.countDown();
+            }
+        });
+
+        config.addListenerConfig(listenerConfig);
+
+        HazelcastInstance instance = hazelcastFactory.newHazelcastInstance(config);
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient();
+        client.shutdown();
+        assertOpenEventually(latch);
```

Comment:
It may be better to use different latches for connect and disconnect. Assume there is a mistake in code and clientConnected is fired twice and clientDisconnected is not fired at all. The test will pass but this is not what we want. you may use the event count logic as you used in the following tests.